### ISSUE TITLE
Python Warning if ipycytoscape extension is not found?

### ISSUE THREAD

**Issue by bdelepine** [QuantStack/ipycytoscape#135]
Hi all,

I would like to suggest the addition of a warning when ipycytoscape fails to find its jupyter notebook extension.

I was unable to run the [Cola notebook example](https://github.com/QuantStack/ipycytoscape/blob/master/examples/Cola%20example.ipynb) (or any other one) as ipycytoscape widget failed to be displayed. There was no Python traceback (nor error, nor warning) when the cell supposed to display the widget was executed (cell 4 with `cytoscapeobj` in Cola notebook example): no output, like a regular cell. On Binder or any working environment, this cell would return the interactive network widget.

I noticed a bunch of errors in the Javascript console when I ran the notebook. It seemed that the widget failed to instantiate: `Could not instantiate widget (anonymous) @ manager-base.js:273`. In the Jupyter console too, there was a `404 GET /nbextensions/jupyter-cytoscape/extension.js`.

The problem was that I use [jupyter_conda extension](https://github.com/fcollonval/jupyter_conda) to manage project-specific kernels within a globally-installed Jupyter notebook: it automatically registers conda environments as suitable kernels to be used by a global jupyter notebook installation. Importantly, Jupyter notebook _is not installed within the environment that owns the kernel_. This means that ipycytoscape notebook extension was not installed at the right place when I followed the installation steps... which explains why it was not found (no widget displayed).

The obvious solution is to start jupyter notebook from within the conda environment in which ipycytoscape and its extension were installed, or to install ipycytoscape directly the base environment that owns jupyter notebook (which fails the purpose of jupyter_conda).

I believe this kind of problem would be easier to diagnostic if ipycytoscape would display a warning when it cannot find its extension (or a message like [bokeh](https://github.com/bokeh/bokeh) does when `output_notebook()` is called).

HTH

**Comment by SylvainCorlay** [comment]
@bdelepine this is a great summary. We have looked into setting up a placeholder widget when a widget module is missing.

This is currently done in a PR to Voilà, but we think this could become core to ipywidget in 8.0.

**Comment by marimeireles** [comment]
So, we shouldn't worry about this now and wait for the ipywidget fix? Or is it worth it to look what you did on voila and reproduce it here?

**Comment by SylvainCorlay** [comment]
I don't think this is in scope for ipycytoscape to fix this, as this is a general issue with widget libraries.

**Comment by bdelepine** [comment]
I agree that a warning handled at the ipywidget level would fix the issue (and would be more elegant).

In the meantime, maybe we could mention something in ipycytoscape install instructions to make sure users are aware that a jupyter notebook extension will be installed along the ipycytoscape package?  This is kind of obvious in retrospect... but I clearly missed that part when I first installed ipycytoscape.

**Comment by marimeireles** [comment]
> In the meantime, maybe we could mention something in ipycytoscape install instructions to make sure users are aware that a jupyter notebook extension will be installed along the ipycytoscape package?

That's a great idea @bdelepine, are you interested in making this contribution yourself? Since you had the problem maybe you can explain it better?

It'd be much appreciated.
Thanks for the bug report :)

**Comment by ianhi** [comment]
This looks like it will become part of ipywidgets core: https://github.com/jupyter-widgets/ipywidgets/pull/2960
